addon/utils.js

```diff
@@ -23,26 +24,30 @@ export let objectAssign = Object.assign || function objectAssign(target) {
 };
 
 /**
-  determine param to set on infinityModel
-  if user passes null, then don't send query param in request
-  if user does not pass anything for value, then see if defined on route
-  else set to default param
-  @method paramsCheck
-  @param {String} value - param passed with infinityRoute
-  @param {String} option - property defined on user route
-  @param {String} - default
-  @return {String}
-*/
-export function paramsCheck(optionParam, defaultParam) {
-  if (typeOf(optionParam) === 'null') {
+ * determine param to set on infinityModel
+ * if user passes null, then don't send query param in request
+ * if user passes option, use it
+ * else set to default param
+ *
+ * @method paramsCheck
+ * @param {String} key - param name
+ * @param {Object} options - parameter overrides
+ * @param {Object} extendedInfinityModel - custom infinity model
+ * @return {String} parameter value
+ */
+export function paramsCheck(key, options, extendedInfinityModel) {
+  const paramDefault = get(extendedInfinityModel.proto(), key);
```

Comment:
Would `get(extendedInfinityModel, key);` w/o the `proto` method still work?

Reply:
Members like _perPageParam_ are defined on the class...should it work without _.proto()_? This is before the instance is created.

Reply:
I think a test will show us our answer 😄 

You may be right though.

Reply:
@snewcomer: I'd [started on a test](https://github.com/ember-infinity/ember-infinity/pull/337/files#diff-24a65c3e3a9ddbe5e79c4a171552f6ba), but wasn't sure how all the pieces fit together. How can I set up mirage to respond to a request with the `?limit=50` parameter?